boards/weact/bluepillplus_ch32v203/doc/index.rst

```diff
@@ -0,0 +1,115 @@
+.. zephyr:board:: bluepillplus_ch32v203
+
+Overview
+********
+
+The `WeActStudio`_ Blue Pill Plus CH32V203 hardware provides support for QingKe 32-bit RISC-V4B
+processor and the following devices:
+
+* CLOCK
+* :abbr:`GPIO (General Purpose Input Output)`
+* :abbr:`NVIC (Nested Vectored Interrupt Controller)`
+
+The board is equipped with two LEDs and two Buttons.
+The `WCH webpage on CH32V203`_ contains the processor's manuals.
+The `WeActStudio webpage on BPP`_ contains the Blue Pill's schematic.
+
+Hardware
+********
+
+The QingKe 32-bit RISC-V4B processor of the Blue Pill Plus CH32V203 is clocked by an external
+8 MHz crystal or the internal 8 MHz oscillator and runs up to 144 MHz.
+The CH32V203 SoC Features 2-4 USART, 4 GPIO ports, 1-2 SPI, 0-2 I2C, 9-16 ADC, RTC,
+CAN, USB Device, USB Host, OPA, and several timers.
+
+Supported Features
+==================
+
+The ``bluepillplus_ch32v203`` board target supports the following hardware features:
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| CLOCK     | on-chip    | clock_control        |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+| PINCTRL   | on-chip    | pinctrl              |
++-----------+------------+----------------------+
+| TIMER     | on-chip    | timer                |
++-----------+------------+----------------------+
+| UART      | on-chip    | uart                 |
++-----------+------------+----------------------+
+
+Other hardware features have not been enabled yet for this board.
```

Comment:
```suggestion
.. zephyr:board-supported-hw::
```

Reply:
So i have a question about this new method, it's based on the supported section in the relevant baord yaml right? Does it enable clock, pinctrol, timer, and uart by default or do i need to add those to the list?

Reply:
`supported` in the board's yaml file is for testing (basically indicating to twister whether a board supports a given feature without it having to actually compile and get the board's devicetree in order to know that)

The documentation table is actually using the board's dts so everything should automagically show up :)

Reply:
Great! Thank you

Reply:
Done